File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int mx = -1, mn = -1, diff = -1;
        for(int i = 0; i<arrays.size(); i++) {
            int currMax = arrays[i][arrays[i].size() - 1];
            int currMin = arrays[i][0];
            if(i == 0) {
                mx = currMax;
                mn = currMin;
            }
            else {
                diff = max(diff, currMax - mn);
                diff = max(diff, mx - currMin);
                mx = max(mx, currMax);
                mn = min(mn, currMin);
            }
        }

        return diff;
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution(object):
    def maxDistance(self, arrays):
        mx = arrays[0][-1]  
        mn = arrays[0][0]   
        diff = 0

        for i in range(1, len(arrays)):
            curr_max = arrays[i][-1]
            curr_min = arrays[i][0]

            diff = max(diff, abs(mx - curr_min))
            diff = max(diff, abs(curr_max - mn))

            mx = max(mx, curr_max)
            mn = min(mn, curr_min)

        return diff
        